function-images/tests/stacking-training/driver/main.py

```diff
@@ -0,0 +1,351 @@
+    X, y = datasets.make_classification(n_samples,
+                                        n_features,
+                                        n_redundant=0,
+                                        n_clusters_per_class=2,
+                                        weights=[0.9, 0.1],
+                                        flip_y=0.1,
+                                        random_state=42)
+    return {'features': X, 'labels': y}
+
+
+def reduce(training_responses) -> dict:
+    log.info("Invoke Reducer")
+    channel = grpc.insecure_channel(args.rAddr)
+    stub = stacking_pb2_grpc.ReducerStub(channel)
+
+    model_keys = []
+    pred_keys = []
+
+    req = stacking_pb2.ReduceRequest()
+    for resp in training_responses:
+        model_pred_tuple = stacking_pb2.ModelPredTuple()
+        model_pred_tuple.model_key = resp['model_key']
+        model_pred_tuple.pred_key = resp['pred_key']
+
+        req.model_pred_tuples.append(model_pred_tuple)
+
+    resp = stub.Reduce(req)
+
+    return {
+        'meta_features_key': resp.meta_features_key,
+        'models_key': resp.models_key
+    }
+
+
+class GreeterServicer(helloworld_pb2_grpc.GreeterServicer):
+    def __init__(self, transferType, XDTconfig=None):
+
+        self.benchName = 'vhive-stacking'
+        self.dataset = generate_dataset()
+        self.modelConfig = model_config
+        self.transferType = transferType
+        if transferType == S3:
+            self.s3_client = boto3.resource(
+                service_name='s3',
+                region_name=os.getenv("AWS_REGION", 'us-west-1'),
+                aws_access_key_id=AWS_ID,
+                aws_secret_access_key=AWS_SECRET
+            )
+        elif transferType == XDT:
+            if XDTconfig is None:
+                log.fatal("Empty XDT config")
+            self.XDTconfig = XDTconfig
+
+    def put(self, obj, key):
+        msg = "Driver uploading object with key '" + key + "' to " + self.transferType
+        log.info(msg)
+        with tracing.Span(msg):
+            pickled = pickle.dumps(obj)
+            if self.transferType == S3:
+                self.put_to_s3(pickled, key)
+            elif self.transferType == XDT:
+                log.fatal("XDT is not supported")
+        return key
+
+    def get(self, key):
+        msg = "Driver gets key '" + key + "' from " + self.transferType
+        log.info(msg)
+        with tracing.Span(msg):
+            response = None
+            if self.transferType == S3:
+                response = self.get_from_s3(key)
+            elif self.transferType == XDT:
+                log.fatal("XDT is not yet supported")
+            return pickle.loads(response['Body'].read())
+    
+    def put_to_s3(self, pickled, key):
+        s3object = self.s3_client.Object(bucket_name=self.benchName, key=key)
+        s3object.put(Body=pickled)
+
+    def get_from_s3(self, key):
+        obj = self.s3_client.Object(bucket_name=self.benchName, key=key)
+        return obj.get()
```

Comment:
these functions are very short, let's inline them into put() and get(). Same for other functions